enable/viewport.py

```diff
@@ -74,10 +73,16 @@ class Viewport(Component):
     #------------------------------------------------------------------------
 
     def __init__(self, **traits):
+        view_position = traits.pop('view_position', None)
+        view_bounds = traits.pop('view_bounds', None)
         Component.__init__(self, **traits)
         # can't use a default because need bounds to be set first
-        if 'view_position' not in traits and self.component is not None:
+        if self.component is not None:
```

Comment:
Because part of the problem we were experienced seemed to be caused by being initialized into an indeterminate state.  This change ensures that to position starts at [0, 0] and then gets properly shifted if `view_position` is not None below.

More computation, but the progression of state is clearer and easier to analyze for debugging.